apps/dashboard/src/utils/code-snippets.ts

```diff
@@ -19,12 +19,18 @@ const safeParsePayload = (payload: string) => {
 
 export const createNodeJsSnippet = ({ identifier, to, payload, secretKey }: CodeSnippet) => {
   const renderedSecretKey = secretKey ? `'${secretKey}'` : `process.env['${SECRET_KEY_ENV_KEY}']`;
-  const euServerUrl = IS_EU ? `,\n  serverURL: 'https://eu.api.novu.co'` : '';
+  let serverUrl = '';
+
+  if (IS_EU) {
+    serverUrl = `,\n  serverURL: 'https://eu.api.novu.co'`;
+  } else if (IS_SELF_HOSTED) {
+    serverUrl = `,\n  serverURL: '${API_HOSTNAME}'`;
+  }
 
   return `import { Novu } from '@novu/api'; 
 
 const novu = new Novu({ 
-  secretKey: ${renderedSecretKey}${euServerUrl}
+  secretKey: ${renderedSecretKey}${serverUrl}
```

Comment:
What about your fancy solution with JSON.stringift? :P

Reply:
In the end, it felt more complicated! 
of of the reasons is the the `variables` would render as string, meaning:
```
secretKey : "process.env.SECRET_KEY_ENV_KEY"
```